tests/views/test_instructor_file.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the `fonzie` instructor_file module.
+"""
+# pylint: disable=no-member,import-error
+from __future__ import absolute_import, unicode_literals
+
+import hashlib
+
+import six
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from django.core.urlresolvers import reverse
+
+from student.roles import CourseStaffRole
+from student.tests.factories import UserFactory
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class InstructorFilesViewTestCase(ModuleStoreTestCase, APITestCase):
+    """Tests for the InstructorFilesView API endpoint"""
+
+    def setUp(self):
+        """
+        Set common parameters for unit tests.
+
+        """
+        super(InstructorFilesViewTestCase, self).setUp()
+
+        self.filename = "fun_test_00_student_profile_info_2019-04-19-1406.csv"
+        self.course = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
+        self.course_staff = UserFactory(
+            username="course_staff", email="course_staff@fun-mooc.fr"
+        )
+        CourseStaffRole(self.course.id).add_users(self.course_staff)
+        self.course_sha1 = hashlib.sha1(six.text_type(self.course.id)).hexdigest()
+
+        self.url = reverse(
+            "fonzie:protected:instructor_file",
+            kwargs={
+                "version": "1.0",
+                "course_sha1": self.course_sha1,
+                "filename": self.filename,
+            },
+        )
```

Comment:
About this section, @sampaccoud told me once that this pattern is weak, _i.e._ you should rather define what you need for each test. Each test should be self-sufficient and understandable without requiring to read the `setup()` method to fully understand what it does. Even if you repeat yourself, it worth it. I am convinced that it's a good idea now.